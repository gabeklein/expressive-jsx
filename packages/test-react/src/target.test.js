const test = require("./_adapter");

test("Simple modifier", `
  () => {
    Foo: {
      color: red;
    }

    <Foo />
  };
`)

test("Attribute-based modifier", `
  () => {
    Foo: {
      background: red;
    }

    hasWhiteText: {
      color: white;
    }

    <Foo hasWhiteText />
  };
`)

test("Nested modifier", `
  () => {
    Foo: {
      color: red;
    }

    Bar: {
      Foo: {
        color: blue;
      }
    }

    <this>
      <Foo />
      <Bar>
        <Foo />
      </Bar>
    </this>
  };
`)

test("Explicitly use modifier", `
  () => {
    Foo: {
      color: blue;
    }

    Baz: {
      use: Foo;
      color: red;
    }

    <this>
      <Baz />
      <Foo />
    </this>
  }
`)

test("collapse single-use modifiers", `
  Bar: { color: red }

  () => {
    Bar: { background: white }
    Foo: {
      Bar: { border: red }
    }
    
    <Foo>
      <Bar />
    </Foo>
  }
`);
