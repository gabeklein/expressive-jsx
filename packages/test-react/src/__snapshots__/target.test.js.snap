// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribute-based modifier 1`] = `
"() => create(Foo, {
  className: \\"hasWhiteText_4b3 Foo_16j\\"
});

CSS.put(\`
	/* 2 */
	.Foo_16j { background: red }
	/* 3 */
	.hasWhiteText_4b3 { color: white }
\`);"
`;

exports[`Attribute-based modifier 2`] = `
"() => <Foo className=\\"hasWhiteText_4b3 Foo_16j\\" />;

CSS.put(\`
	/* 2 */
	.Foo_16j { background: red }
	/* 3 */
	.hasWhiteText_4b3 { color: white }
\`);"
`;

exports[`Explicitly use modifier 1`] = `
"() => create(Fragment, {}, create(Baz, {
  className: \\"Foo_76l Baz_16j\\"
}), create(Foo, {
  className: \\"Foo_76l\\"
}));

CSS.put(\`
	/* 2 */
	.Baz_16j { color: red }
	/* 3 */
	.Foo_76l { color: blue }
\`);"
`;

exports[`Explicitly use modifier 2`] = `
"() => <Fragment><Baz className=\\"Foo_76l Baz_16j\\" /><Foo className=\\"Foo_76l\\" /></Fragment>;

CSS.put(\`
	/* 2 */
	.Baz_16j { color: red }
	/* 3 */
	.Foo_76l { color: blue }
\`);"
`;

exports[`Nested modifier 1`] = `
"() => create(Fragment, {}, create(Foo, {
  className: \\"Foo_76l\\"
}), create(Bar, {}, create(Foo, {
  className: \\"Foo_76l Foo_6n9\\"
})));

CSS.put(\`
	/* 2 */
	.Foo_6n9 { color: blue }
	/* 3 */
	.Foo_76l { color: red }
\`);"
`;

exports[`Nested modifier 2`] = `
"() => <Fragment><Foo className=\\"Foo_76l\\" /><Bar><Foo className=\\"Foo_76l Foo_6n9\\" /></Bar></Fragment>;

CSS.put(\`
	/* 2 */
	.Foo_6n9 { color: blue }
	/* 3 */
	.Foo_76l { color: red }
\`);"
`;

exports[`Simple modifier 1`] = `
"() => create(Foo, {
  className: \\"Foo_16j\\"
});

CSS.put(\`
	/* 2 */
	.Foo_16j { color: red }
\`);"
`;

exports[`Simple modifier 2`] = `
"() => <Foo className=\\"Foo_16j\\" />;

CSS.put(\`
	/* 2 */
	.Foo_16j { color: red }
\`);"
`;

exports[`collapse single-use modifiers 1`] = `
"() => create(Foo, {}, create(Bar, {
  className: \\"Bar_i3g\\"
}));

CSS.put(\`
	/* 2 */
	.Bar_i3g { color: red; border: red solid 1px; background: white }
\`);"
`;

exports[`collapse single-use modifiers 2`] = `
"() => <Foo><Bar className=\\"Bar_i3g\\" /></Foo>;

CSS.put(\`
	/* 2 */
	.Bar_i3g { color: red; border: red solid 1px; background: white }
\`);"
`;
